x-pack/filebeat/module/aws/vpc/ingest/pipeline.yml

```diff
@@ -0,0 +1,30 @@
+description: "Pipeline for s3 server access logs"
+
+processors:
+  - grok:
```

Comment:
I think `dissect` could be used on Beats side to parse this line. This would have the benefit of allowing the use of the `community_id` processor which is important for correlating flows with other events.

So if dissect can be used for this message the processor chain would look something like:

```
processors:
- dissect:
    field: message
    target_prefix: aws.vpc
    tokenizer: '%{version} %{account_id} %{interface_id}' ...
- convert:
    - {from: aws.vpc.srcaddr, to: source.address}
    - {from: aws.vpc.srcaddr, to: source.ip, type: ip}
    - {from: aws.vpc.srcport, to: source.port, type: long}
    - {from: aws.vpc.dstaddr, to: destination.address}
    - {from: aws.vpc.dstaddr, to: destination.ip, type: ip}
    - {from: aws.vpc.dstport, to: destination.port, type: long}
    - {from: aws.vpc.protocol, to: network.iana_number, type: long}
    - {from: aws.vpc.packets, to: source.packets, type: long}
    - {from: aws.vpc.bytes, to: source.bytes, type: long}
    - {from: aws.vpc.packets, to: network.packets, type: long}
    - {from: aws.vpc.bytes, to: network.bytes, type: long}
- community_id: ~
- timestamp:
    layouts: [UNIX]
    field: aws.vpc.start
    target: event.start
- timestamp:
    layouts: [UNIX]
    field: aws.vpc.end
    target: event.end

# Use the aws.vpc.action value to set the event.outcome value to either "allow" or "deny".
- add_fields:
     when.equals.aws.vpc.action: PERMIT
     target: event
     fields: {outcome: allow}
- add_fields:
     when.equals.aws.vpc.action: REJECT
     target: event
     fields: {outcome: deny}
```

Then in your Ingest Node pipeline you can do the geoip and ASN enrichment. (Basically copy/paste from https://github.com/elastic/beats/blob/master/x-pack/filebeat/module/cisco/ios/ingest/pipeline.yml.)

Reply:
Thank you for the review!! The commit I just pushed is just half working. I have all the processors in place but timestamp is not converted correctly from `aws.vpc.start` and `aws.vpc.end` for some reason. geoip and ASN is not working either. Will investigate more!

Reply:
Done, both are working now.